server/utils/activityHooks.ts

```diff
@@ -0,0 +1,34 @@
+import { defer } from 'server/utils/deferred';
+
+type CreateActivityHooksOptions<Model> = {
+	Model: any;
```

Comment:
Could this be like

```ts
import { Model as SequelizeModel } from 'sequelize'

type CreateActivityHooksOptions<Model> = {
  Model: typeof SequelizeModel
}
```

instead of `any`?

Reply:
This would be a helpful clue to the reader, but I don't know if it would provide any type safety, since TS treats all the model singletons as `any`. Maybe there's a `@types/sequelize` that could help us here?

Reply:
This could be totally wrong, but I was thinking that if we do `typeof SequelizeModel`, it would capture the type of the complete model class. I tried something that seems to kinda work below... but it's funky. We'd at least get type checking for stuff like `afterCreate` etc. I'm sure there's a better way to do this.

<img width="500" alt="Screen Shot 2021-07-27 at 3 41 34 PM" src="https://user-images.githubusercontent.com/6402908/127217426-7b6cefad-dfc8-411e-a119-e70fe5929bc0.png">

Reply:
Ah, that's extremely useful to know about! In some other branch I have been building up my own `SequelizeModel<T>` that augments `T` with ORM methods like `.toJSON()` It seems I might be able to use this instead. Two problems remain:

1. The model singleton has the `afterCreate` etc. methods (e.g. `Pub` instead of a `pub`). I don't know if Sequelize has a type for that.
2. `Model` is apparently actually generic: `Model<T, T2>`, but neither type argument seems to provide the instance-specific properties when I try it (i.e. `Model<{id: string}>` doesn't have an `id` property). I think that's a bug.

I did end up adding some better types to this to document our expectations. Feel free to weigh in on these; I'm worried it came out a little overwrought.